### PR TITLE
[docs]: improve workers page (max_builds and formating)

### PR DESCRIPTION
**Pull request by Frodox**
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by Frodox** [comment]
@seankelly please, review it :)

File: master/docs/manual/cfg-workers.rst [hunk 1]
```diff
@@ -47,14 +50,22 @@ For example::
                       properties={ 'os':'solaris' }),
     ]
 
+Limiting Concurrency
+++++++++++++++++++++
+
 .. index:: Workers; limiting concurrency
 
-The :class:`Worker` constructor can also take an optional ``max_builds`` parameter to limit the number of builds that it will execute simultaneously::
+The :class:`Worker` constructor can also take an optional ``max_builds`` parameter which is total number of *different* builders than can run on that worker::
```
Comment (tardyp) [inline master/docs/manual/cfg-workers.rst]: I think the previous sentence was more precise.
max_builds limit the number of concurent builds that happens on a master. By buildbot design, It happens that one worker can only work on one build of the same builder at the same time. But this is not handled at all by the max_builds logic.

Your sentence imply that you can only configure max_builds on the worker, which is not true.

**Comment by Frodox** [comment]
Please, take a look

File: master/docs/manual/cfg-workers.rst [hunk 2]
```diff
@@ -146,14 +157,14 @@ In order to use local workers you need to have ``buildbot-worker`` package insta
...
+The standard Buildbot model has workers started manually.
 The previous section described how to configure the master for this approach.
 
-Another approach is to let the buildbot master start workers when builds are ready, on-demand.
+Another approach is to let the Buildbot master start workers when builds are ready, on-demand.
 Thanks to services such as Amazon Web Services' Elastic Compute Cloud ("AWS EC2"), this is relatively easy to set up, and can be very useful for some situations.
 
 The workers that are started on-demand are called "latent" workers.
-As of this writing, buildbot ships with an abstract base class for building latent workers, and a concrete implementation for AWS EC2 and for libvirt.
+As of this writing, Buildbot ships with an abstract base class for building latent workers, and a concrete implementation for AWS EC2 and for libvirt.
```
Comment (seankelly) [inline master/docs/manual/cfg-workers.rst]: Plus Docker, OpenStack, Hyper, and Marathon.

File: master/docs/manual/cfg-workers.rst [hunk 3]
```diff
@@ -47,14 +50,22 @@ For example::
...
 
     c['workers'] = [
-        worker.Worker("bot-linux", "linuxpassword", max_builds=2)
+        worker.Worker('bot-linux', 'linuxpassword',
+                      max_builds=2),
     ]
 
+.. note::
+
+    In :ref:`worker-for-builders` concept only one build from same builder would run on the worker.
```
Comment (seankelly) [inline master/docs/manual/cfg-workers.rst]: Add "the" before "same".

**Comment by Frodox** [comment]
@seankelly updated

**Comment by seankelly** [comment]
LGTM.

**Pull request merged**